domain.go

```diff
@@ -3,16 +3,35 @@ package generictypes
 import (
 	"encoding/json"
 	"log"
+	"sync"
 
 	"github.com/giantswarm/validate"
 	"github.com/giantswarm/validate/web"
 	"github.com/juju/errgo"
 )
 
-func init() {
-	// Fetch a new list of TLDs from the internet on startup.
-	if err := web.UpdateTLDs(web.IANA); err != nil {
-		log.Printf("[ERROR] Failed to update TLDs: %v\n", err)
+var (
+	updatedTLDs    bool
+	updateTLDMutex sync.Mutex
+)
+
+// updateTLDsIfNeeded performs a webrequest to update the
+// list of toplevel domain names.
+// This is done only once per process.
+func updateTLDsIfNeeded() {
+	if updatedTLDs {
+		return
+	}
+
+	updateTLDMutex.Lock()
+	defer updateTLDMutex.Unlock()
+
+	if !updatedTLDs {
+		// Fetch a new list of TLDs from the internet on startup.
+		if err := web.UpdateTLDs(web.IANA); err != nil {
+			log.Printf("[ERROR] Failed to update TLDs: %v\n", err)
```

Comment:
Here should be real error handling. The caller should decide what to do in this case. Further when there is an error here, `updatedTLDs` should not be set to `true`.